src/main/file_util.java

```diff
@@ -161,24 +177,21 @@ public static void readglobalHeader(RandomAccessFile file_stream,int offset,Map<
 		file_stream.seek(0);
 		file_stream.read(globalHeader, 0, offset);
 		
-		header_parser.parse(globalHeader,header_table);
-		
+		header_parser.parse(globalHeader,header_table);		
 	}
 
 	public static void loadFile(File file,ByteArrayOutputStream outputStream,Map<String,String> header_table) throws IOException {
-		
-		RandomAccessFile file_stream = new RandomAccessFile(file, "r");
-		
-		int offset = calcPyloadStart(file_stream);
-
-		readglobalHeader(file_stream,offset,header_table);
-		file_stream.seek(offset);
-		int count = 0;
-		byte[] buffer = new byte[1024];
-		while ( (count = file_stream.read(buffer)) !=-1) {
-			outputStream.write(buffer,0,count);
-		}
-		file_stream.close();

```

Comment:
file stream not closed?

Reply:
It gets closed automatically as soon as the try-with-resource statement ends. It is equivalend to the using statement for c# if you are familiar with that one.